x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/dimension_panel.tsx

```diff
@@ -137,6 +138,10 @@ export const IndexPatternDimensionPanel = memo(function IndexPatternDimensionPan
                 field: droppedItem.field,
               });
 
+          trackUiEvent('drop_onto_dimension');
+          const hasData = Object.values(props.state.layers).some(({ columns }) => columns.length);
+          trackUiEvent(hasData ? 'drop_non_empty' : 'drop_empty');
```

Comment:
This is fine, but also, do we need three different events? drop_onto_dimension would be a sum of drop_non_empty and drop_empty

Reply:
It's actually five events:

`drop_total` is every drop

Two pairs:

`drop_onto_workspace` + `drop_onto_dimension` = `drop_total`
`drop_empty` + `drop_non_empty` = `drop_total`